Sources/Sentry/SentryClient.m

```diff
@@ -472,13 +475,46 @@ - (void)captureSession:(SentrySession *)session
     }
 
     SentryEnvelopeItem *item = [[SentryEnvelopeItem alloc] initWithSession:session];
-    SentryEnvelopeHeader *envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:nil
-                                                                       traceContext:nil];
-    SentryEnvelope *envelope = [[SentryEnvelope alloc] initWithHeader:envelopeHeader
+    SentryEnvelope *envelope = [[SentryEnvelope alloc] initWithHeader:[SentryEnvelopeHeader empty]
                                                            singleItem:item];
     [self captureEnvelope:envelope];
 }
 
+- (void)captureReplayEvent:(SentryReplayEvent *)replayEvent
+           replayRecording:(SentryReplayRecording *)replayRecording
+                     video:(NSURL *)videoURL
+                 withScope:(SentryScope *)scope
+{
+    replayEvent = (SentryReplayEvent *)[self prepareEvent:replayEvent
+                                                withScope:scope
+                                   alwaysAttachStacktrace:NO];
+
+    if (![replayEvent isKindOfClass:SentryReplayEvent.class]) {
+        SENTRY_LOG_DEBUG(@"The event preprocessor didn't update the replay event in place. The "
+                         @"replay was discarded.");
+        return;
+    }
+
+    // breadcrumbs for replay will be send with ReplayRecording
+    replayEvent.breadcrumbs = nil;
```

Comment:
we probably also got to nullify other stuff that replays do not support currently:
```
replayEvent.extra = nil;
replayEvent.threads = nil;
replayEvent.debugMeta = nil;
```

Reply:
and also this part which will be executed for the replay event I believe, but it shouldn't: https://github.com/getsentry/sentry-cocoa/blob/7b5ebd9f178fc217ef715d615db466d5d77c52ec/Sources/Sentry/SentryClient.m#L557-L562

Reply:
Good catch. I will fix this.